reproducibility_project/src/engines/gromacs/project.py

```diff
@@ -19,171 +16,174 @@ def __init__(self):
-def nvt_grompp(job):
-    """Label for the grompp_nvt step."""
-    return job.isfile("nvt.tpr")
-
-
-@Project.label
-def nvt_completed(job):
-    """Label for the gmx_nvt step."""
-    return job.isfile("nvt.gro")
-
-
-@Project.label
-def npt_grompp(job):
-    """Label for the grompp_npt step."""
-    return job.isfile("npt.tpr")
-
-
-@Project.label
-def npt_completed(job):
-    """Label for the gmx_npt step."""
-    return job.isfile("npt.gro")
-
-
-"""Setting up workflow operation"""
-
-
 @Project.operation
-@Project.pre(lambda j: j.sp.simulation_engine == "gromacs")
+@Project.pre(lambda j: j.sp.engine == "gromacs")
+@Project.post(lambda j: j.isfile("init.gro"))
+@Project.post(lambda j: j.isfile("init.top"))
+@Project.post(lambda j: j.isfile("em.mdp"))
+@Project.post(lambda j: j.isfile("nvt.mdp"))
+@Project.post(lambda j: j.isfile("npt.mdp"))
+@flow.with_job
 def init_job(job):
     """Initialize individual job workspace, including mdp and molecular init files."""
-    from project.src.molecules.system_builder import construct_system
-
-    with job:
-        # Create a Compound and save to gro and top files
-        system = construct_system(job.sp)
-        system.save(filename="init.gro")
-        system.save(filename="init.top", forcefield_name=job.sp.forcefield_name)
-
-        # Modify mdp files according to job statepoint parameters
-        mdp_abs_path = os.path.dirname(os.path.abspath(mdp.__file__))
-        mdps = {
-            "em": {
-                "fname": "em.mdp",
-                "template": f"{mdp_abs_path}/em_template.mdp.jinja",
-                "data": dict(),
+    sys.path.append(Project().root_directory() + "/..")
+    from reproducibility_project.src.engine_input.gromacs import mdp
+    from reproducibility_project.src.molecules.system_builder import (
+        construct_system,
+    )
+
+    # Create a Compound and save to gro and top files
+    system = construct_system(job.sp)
+    system[0].save(filename="init.gro", overwrite=True)
+    if job.sp.forcefield_name in ["oplsaa", "trappe-ua"]:
+        system[0].save(
+            filename="init.top",
+            forcefield_name=job.sp.forcefield_name,
+            overwrite=True,
+        )
+    elif job.sp.forcefield_name == "spce":
+        from reproducibility_project.src import xmls
+
+        ff_path = (
+            str(os.path.dirname(os.path.abspath(xmls.__file__))) + "/spce.xml"
+        )
+        system[0].save(
+            filename="init.top", forcefield_files=ff_path, overwrite=True
+        )
+    elif job.sp.forcefield_name == "benzene-ua":
+        from reproducibility_project.src import xmls
+
+        ff_name = "benzene_trappe-ua_like.xml"
+        ff_path = (
+            str(os.path.dirname(os.path.abspath(xmls.__file__))) + "/" + ff_name
+        )
+        system[0].save(
+            filename="init.top", forcefield_files=ff_path, overwrite=True
+        )
+
+    # Modify mdp files according to job statepoint parameters
+    import unyt as u
```

Comment:
Lets move this to the top of the file with the other imports.

Reply:
Fixed